src/solvers/dg_curved/containers_2d.jl

```diff
@@ -0,0 +1,91 @@
+function init_elements!(elements, mesh::CurvedMesh{2}, basis::LobattoLegendreBasis{T, NNODES}) where {T, NNODES}
+  @unpack faces = mesh
+  @unpack node_coordinates, left_neighbors, metric_terms, inverse_jacobian = elements
+
+  linear_indices = LinearIndices(size(mesh))
+
+  # Calculate inverse Jacobian and node coordinates
+  for cell_x in 1:size(mesh, 1), cell_y in 1:size(mesh, 2)
+    element = linear_indices[cell_x, cell_y]
+
+    calc_node_coordinates!(node_coordinates, element, cell_x, cell_y, mesh, basis)
+
+    calc_metric_terms!(metric_terms, element, node_coordinates, basis)
+
+    calc_inverse_jacobian!(inverse_jacobian, element, metric_terms)
+  end
+
+  initialize_neighbor_connectivity!(left_neighbors, mesh, linear_indices)
+
+  return nothing
+end
+
+
+function calc_node_coordinates!(node_coordinates, element,
+                                cell_x, cell_y,
+                                mesh::CurvedMesh{2},
+                                basis::LobattoLegendreBasis{T, NNODES}) where {T, NNODES}
+  @unpack nodes = basis
+
+  # Get cell length in reference mesh
+  dx = 2 / size(mesh, 1)
+  dy = 2 / size(mesh, 2)
+
+  # Calculate node coordinates of reference mesh
+  cell_x_offset = -1 + (cell_x-1) * dx + dx/2
+  cell_y_offset = -1 + (cell_y-1) * dy + dy/2
+
+  for j in 1:NNODES, i in 1:NNODES
+    # node_coordinates are the mapped reference node_coordinates
+    node_coordinates[:, i, j, element] .= transfinite_mapping(cell_x_offset + dx/2 * nodes[i],
+                                                              cell_y_offset + dy/2 * nodes[j], mesh)
+  end
+end
+
+
+function calc_metric_terms!(metric_terms, element, node_coordinates::Array{RealT, 4}, 
+                            basis::LobattoLegendreBasis{T, NNODES}) where {RealT, T, NNODES}
+  @views metric_terms[1, 1, :, :, element] .= basis.derivative_matrix * node_coordinates[1, :, :, element] # x_ξ
+  @views metric_terms[2, 1, :, :, element] .= basis.derivative_matrix * node_coordinates[2, :, :, element] # y_ξ
+  @views metric_terms[1, 2, :, :, element] .= node_coordinates[1, :, :, element] * basis.derivative_matrix' # x_η
+  @views metric_terms[2, 2, :, :, element] .= node_coordinates[2, :, :, element] * basis.derivative_matrix' # y_η
```

Comment:
You could even use the inplace versions to avoid creating a temporary array.
```suggestion
  @views mul!(metric_terms[1, 1, :, :, element], basis.derivative_matrix, node_coordinates[1, :, :, element]) # x_ξ
  @views mul!(metric_terms[2, 1, :, :, element], basis.derivative_matrix, node_coordinates[2, :, :, element]) # y_ξ
  @views mul!(metric_terms[1, 2, :, :, element], node_coordinates[1, :, :, element], basis.derivative_matrix') # x_η
  @views mul!(metric_terms[2, 2, :, :, element], node_coordinates[2, :, :, element], basis.derivative_matrix') # y_η
```

Reply:
Very nice! Is there anything similar we can do with `calc_inverse_jacobian!`?
```julia
function calc_inverse_jacobian!(inverse_jacobian::AbstractArray{<:Any, 3}, element, metric_terms)
  @views inverse_jacobian[:, :, element] = inv.(metric_terms[1, 1, :, :, element] .* metric_terms[2, 2, :, :, element] .-
                                                metric_terms[1, 2, :, :, element] .* metric_terms[2, 1, :, :, element])

  return inverse_jacobian
end
```

Reply:
You just need one dot more - `.=` instead of `=`. However, I would prefer to use `@.` to make this more clear, i.e.
```julia
function calc_inverse_jacobian!(inverse_jacobian::AbstractArray{<:Any, 3}, element, metric_terms)
  @. @views inverse_jacobian[:, :, element] = inv(metric_terms[1, 1, :, :, element] * metric_terms[2, 2, :, :, element] -
                                                  metric_terms[1, 2, :, :, element] * metric_terms[2, 1, :, :, element])

  return inverse_jacobian
end
```
But you should benchmark this.

Reply:
(by the way, I would be interested to see the performance difference between your version with `*` and the version with `mul!`)

Reply:
```
mul!:
BenchmarkTools.Trial: 
  memory estimate:  896 bytes
  allocs estimate:  16
  --------------
  minimum time:     785.859 ns (0.00% GC)
  median time:      806.061 ns (0.00% GC)
  mean time:        918.035 ns (10.19% GC)
  maximum time:     84.262 μs (98.55% GC)
  --------------
  samples:          10000
  evals/sample:     99

what we had before:
BenchmarkTools.Trial:
  memory estimate:  1.69 KiB
  allocs estimate:  20
  --------------
  minimum time:     958.824 ns (0.00% GC)
  median time:      1.018 μs (0.00% GC)
  mean time:        1.225 μs (11.98% GC)
  maximum time:     381.141 μs (99.45% GC)
  --------------
  samples:          10000
  evals/sample:     17